### ISSUE TITLE
Systems Test driven_by don't pass additional options to drivers other than Selenium

### ISSUE THREAD

**Issue by mariochavez** [rails/rails#29000]
### Steps to reproduce

1. Create a Rails 5.1 application
2. Add poltergeist gem (or capybara-webkit)
3. Generate a scaffold `bin/rails g scaffold user name`
4. Run migrations `bin/rails db:migrate`
5. Create a System Test `bin/rails g system_test sample`
6. Add the following test:
```
  test "visiting the index" do
    visit users_url

    assert_selector "h1", text: "Simple"
  end

```
7. Require poltergeist gem in `spec/test_helper.rb`
8. Change `driven_by` call to use poltergeist and add options as follow:
```
# test/application_system_test_case.rb
driven_by :poltergeist, options: { js_errors: false }
```
9. Modify `application.js` to require itself and to error out
```
//= require_self

console.log($('h1'));
```

### Expected behavior
Tests is expected to run and display outcome while poltergeist ignores the javascript problem.

### Actual behavior
Poltergeist don't receive `driven_by` options and it fails the test with javascript error:
```
SimplesTest#test_visiting_the_index:
Capybara::Poltergeist::JavascriptError: One or more errors were raised in the Javascript code on the page. If you don't care about these errors, you can ignore them by setting js_errors: false in your Poltergeist configuration (see documentation for details).

ReferenceError: Can't find variable: $
ReferenceError: Can't find variable: $
    at http://127.0.0.1:61282/assets/application-451aa84cafd96af60cda12f1ca3ae5d56e26827259b4067584c396216e21680d.js:1327 in global code
    test/system/simples_test.rb:5:in `block in <class:SimplesTest>'
```

### System configuration
**Rails version**: 5.1.0

**Ruby version**: 2.4.1

### Debug
`ActionDispatch::SystemTesting::Driver#use`[1](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/system_testing/driver.rb) method register and pass options correctly only for  `:selenium` driver, for any other drivers it relies on the fact that each one register by them self without options.

Documentation is not clear what happens `options` is passed to `driven_by`, is not even mentioned. Drivers like poltergeist supports options like `screen_size` but is not used.

### Proposals
Should `ActionDispatch::SystemTesting::Driver` be extended to register and pass options to additional drivers? There are not to many drivers available.

Maybe `use` method can be modified to:
```
  def use
    case @name
    when :selenium then register
    when :poltergeist then register_poltergeist
    end

    setup
  end

  private

  def register_poltergeist
    if defined?(Capybara::Poltergeist)
      Capybara.register_driver :poltergeist do |app|
          Capybara::Poltergeist::Driver.new(app, @options.merge(window_size:
          @screen_size))
      end
    end
  end
```

This will allow to configure poltergeist driver as follow:
```
  driven_by :poltergeist, screen_size: [1400, 1400], options: { js_errors: false }
```

if poltergeist gem is not in `Gemfile` or hasn't been required in `test_helper` but someone setup `driven_by` to use poltergeist, Capybara will let the user know what is missing:
```
Capybara::DriverNotFoundError: no driver called :poltergeist was found, available drivers: :rack_test, :selenium
    test/system/simples_test.rb:5:in `block in <class:SimplesTest>'
```

Maybe the same can be done for capybara-webkit.

A second approach could be just documenting that `driven_by` will not pass options param to driver if it is not selenium and if you want to configure the driver the you need to do it inside `ApplicationSystemTestCase` like:
```
class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
  options = { js_errors: false, window: [1400, 1400] }
  Capybara.register_driver :poltergeist do |app|
    Capybara::Poltergeist::Driver.new(app, options)
  end

  driven_by :poltergeist
end
```

Either one of the approaches will save time for users moving to Rails' System Test.

I'll be happy to work on any of these approaches if the core team decides that is something that needs to happen.

**Comment by maclover7** [comment]
related PR #28341, cc @eileencodes

**Comment by eileencodes** [comment]
@mariochavez thanks for the report. It probably does make sense to pass options to the other driver s if applicable. Do you know if `js_errors` is an option for Capybara Webkit as well?

**Comment by mariochavez** [comment]
@eileencodes for Capybara Webkit is called `raise_javascript_errors`. But given than `options` param in `driven_by` is a `Hash`, public API don't need to change, any _arbitrary_ parameter that underneath driver support can be passed in `options` Hash.

**Comment by eileencodes** [comment]
@mariochavez I missed that you said you wanted to work on this. Do you still want to take a stab at fixing this (I prefer adding namespaced register methods and handling cases in the register method)? If not, there's another contributor who wants to work on it. Let me know.

**Comment by mariochavez** [comment]
Hi @eileencodes no worries, sure I'll send a PR for this.

**Comment by mariochavez** [comment]
@eileencodes my PR is ready for review. I tested it with a few real applications using poltergeist/capybara-webkit and works as expected.

**Comment by eileencodes** [comment]
Closed by #29315
